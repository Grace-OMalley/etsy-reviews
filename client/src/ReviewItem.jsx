import React from 'react';
import styles from './styles/ReviewItem.module.css';

class ReviewItem extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      clampText: true
    }
    this.handleClick = this.handleClick.bind(this);
  }

  componentDidUpdate(prevProps) {
    if (prevProps.review._id !== this.props.review._id) {
      this.setState({
        clampText: true
      })
    }
  }

  handleClick() {
    this.setState({
      clampText: !this.state.clampText
    })
  }

  render() {
    const renderStars = () => {
      let rating = this.props.review.starRating;
      const stars = [];
      for (let i = 0; i < 5; i++) {
        if (i < rating) {
          stars.push(<span key={i} className={'solidStar'} data-rating="0"><svg xmlns="http://www.w3.org/2000/svg" viewBox="3 3 18 18" aria-hidden="true" focusable="false"><path d="M20.83,9.15l-6-.52L12.46,3.08h-.92L9.18,8.63l-6,.52L2.89,10l4.55,4L6.08,19.85l.75.55L12,17.3l5.17,3.1.75-.55L16.56,14l4.55-4Z"></path></svg></span>);
        } else {
          stars.push(<span key={i} className={'hollowStar'} data-rating="4"><svg xmlns="http://www.w3.org/2000/svg" viewBox="3 3 18 18" aria-hidden="true" focusable="false"><path d="M12,6.47l1.21,2.84.41,1,1.05.09,3.07.27-2.32,2-.8.69.24,1,.69,3L12.9,15.79l-.9-.53-.9.53L8.45,17.38l.69-3,.24-1-.8-.69-2.32-2,3.07-.27,1.05-.09.41-1L12,6.47m.46-3.39h-.92L9.18,8.63l-6,.52L2.89,10l4.55,4L6.08,19.85l.75.55L12,17.3l5.17,3.1.75-.55L16.56,14l4.55-4-.28-.88-6-.52L12.46,3.08Z"></path></svg></span>);
        }
      }
      return stars;
    }

    return (
      <div className={styles.reviewItem}>

        <div className={styles.userDate} css="user-date">
          <div className={styles.profileContainer} css="user-profile-img-container">
            <img src={this.props.review.userPic} className={styles.profilePic} css="profile-pic"/>
          </div>
          <div className={styles.usernameDate}css="username-date-container">
            <p><a className={styles.anchor}>{this.props.review.username}</a>{this.props.review.reviewDate.slice(4, -42)}</p>
          </div>
        </div>

          <div className={styles.bodyContainer} css="body-text-container">
            <div className={styles.innerBodyContainer} css="inner-body-container">
              <div className={styles.innermostContainer}>
              <div className={styles.innerContainer} css="body-text" css="margin-right-48px">
                <div className={styles.starsContainer} css="margin-left-6px">
                  <span className={styles.starsSpan} css="stars-span">
                    {renderStars()}
                    </span>
                  </div>
                </div>

                <div className={styles.specsContainer} css="specs-container">
                  <div className={styles.innerSpecsContainer} css="inner-specs-container">
                    <ul>
                      {this.props.review.productSpecs.map((spec, i) => (
                        <li className={styles.listItem}>
                        <p className={styles.specType}>Shirt Size: </p>
                        <p className={styles.specBody}>{spec.Size}</p>
                        </li>
                      ))}
                    </ul>
                  </div>
                </div>

                <div className={styles.textBodyContainer} css="text-body-container">
                  <div className={styles.textBody} css="text-body">
                    <div className={styles.text}>
                      <p className ={this.state.clampText === true ? styles.clamped : ''}>{this.props.review.reviewBody}</p>
                    </div>
                    <button className={`${styles.textButton} ${this.state.clampText === false ? styles.hidden : ''}`} onClick = {this.handleClick}></button>
                  </div>
                </div>
              </div>

              {/* <div className="user-img-container">
                <button>
                  <img src={this.props.review.userProductImage} />
                </button>
            </div> */}
          </div>
        </div>
      </div>
    )
  }
}

export default ReviewItem;