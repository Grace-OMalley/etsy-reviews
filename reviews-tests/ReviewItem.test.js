import React from 'react';
import ReviewItem from '../client/src/ReviewItem.jsx'
import data from './testData.js';

// import styles from '../client/src/styles/PageNav.module.css';
import {shallow, mount, render} from 'enzyme';

describe('Review Item component', () => {
  it('Should render solid stars according to user rating', () => {
    const wrapper = mount(<ReviewItem review={data.review} />);
    const solidStars = wrapper.find('.csolidStar');
    expect(solidStars).toHaveLength(data.review.starRating);
  })
})