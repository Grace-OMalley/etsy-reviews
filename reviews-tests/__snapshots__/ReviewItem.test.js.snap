// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review Item should render to the page 1`] = `
<div
  className="reviewItem"
>
  <div
    className="userDate"
    css="user-date"
  >
    <div
      className="profileContainer"
      css="user-profile-img-container"
    >
      <img
        className="profilePic"
        css="profile-pic"
        src="https://cdn.fakercloud.com/avatars/madysondesigns_128.jpg"
      />
    </div>
    <div
      className="usernameDate"
      css="username-date-container"
    >
      <p>
        <a
          className="anchor"
        >
          Kelsie Harvey
        </a>
        Jun 13 2020
      </p>
    </div>
  </div>
  <div
    className="bodyContainer"
    css="body-text-container"
  >
    <div
      className="innerBodyContainer"
      css="inner-body-container"
    >
      <div
        className="innermostContainer"
      >
        <div
          className="innerContainer"
          css="margin-right-48px"
        >
          <div
            className="starsContainer"
            css="margin-left-6px"
          >
            <span
              className="starsSpan"
              css="stars-span"
            >
              <span
                data-rating="0"
              >
                <svg
                  aria-hidden="true"
                  focusable="false"
                  viewBox="3 3 18 18"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M20.83,9.15l-6-.52L12.46,3.08h-.92L9.18,8.63l-6,.52L2.89,10l4.55,4L6.08,19.85l.75.55L12,17.3l5.17,3.1.75-.55L16.56,14l4.55-4Z"
                  />
                </svg>
              </span>
              <span
                data-rating="0"
              >
                <svg
                  aria-hidden="true"
                  focusable="false"
                  viewBox="3 3 18 18"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M20.83,9.15l-6-.52L12.46,3.08h-.92L9.18,8.63l-6,.52L2.89,10l4.55,4L6.08,19.85l.75.55L12,17.3l5.17,3.1.75-.55L16.56,14l4.55-4Z"
                  />
                </svg>
              </span>
              <span
                data-rating="0"
              >
                <svg
                  aria-hidden="true"
                  focusable="false"
                  viewBox="3 3 18 18"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M20.83,9.15l-6-.52L12.46,3.08h-.92L9.18,8.63l-6,.52L2.89,10l4.55,4L6.08,19.85l.75.55L12,17.3l5.17,3.1.75-.55L16.56,14l4.55-4Z"
                  />
                </svg>
              </span>
              <span
                data-rating="0"
              >
                <svg
                  aria-hidden="true"
                  focusable="false"
                  viewBox="3 3 18 18"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M20.83,9.15l-6-.52L12.46,3.08h-.92L9.18,8.63l-6,.52L2.89,10l4.55,4L6.08,19.85l.75.55L12,17.3l5.17,3.1.75-.55L16.56,14l4.55-4Z"
                  />
                </svg>
              </span>
              <span
                data-rating="0"
              >
                <svg
                  aria-hidden="true"
                  focusable="false"
                  viewBox="3 3 18 18"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M20.83,9.15l-6-.52L12.46,3.08h-.92L9.18,8.63l-6,.52L2.89,10l4.55,4L6.08,19.85l.75.55L12,17.3l5.17,3.1.75-.55L16.56,14l4.55-4Z"
                  />
                </svg>
              </span>
            </span>
          </div>
        </div>
        <div
          className="specsContainer"
          css="specs-container"
        >
          <div
            className="innerSpecsContainer"
            css="inner-specs-container"
          >
            <ul>
              <li
                className="listItem"
              >
                <p
                  className="specType"
                >
                  SPEC TYPE: 
                </p>
                <p
                  className="specBody"
                >
                   SPECIFICATION_BODY
                </p>
              </li>
              <li
                className="listItem"
              >
                <p
                  className="specType"
                >
                  SPEC TYPE: 
                </p>
                <p
                  className="specBody"
                >
                   SPECIFICATION_BODY
                </p>
              </li>
            </ul>
          </div>
        </div>
        <div
          className="textBodyContainer"
          css="text-body-container"
        >
          <div
            className="textBody"
            css="text-body"
          >
            <div
              className="text"
            >
              <p>
                Provident optio dolores numquam dolorem quibusdam necessitatibus expedita tenetur mollitia. Fugiat rerum porro et laboriosam voluptas consequatur quas. Itaque nostrum necessitatibus. Voluptatem consequatur amet. Nemo deserunt nam fugit inventore et nam perferendis ut.
 
Et perferendis sunt enim tempore. Quasi earum necessitatibus soluta. Et ea minus veritatis incidunt ipsa facere nostrum aliquid itaque. Laudantium debitis dolores. Soluta fugiat ut.
 
Voluptatem voluptas totam in iste omnis harum quidem. Molestiae eos et at sed et et. Aliquam et possimus pariatur ex blanditiis. Laborum in doloribus laboriosam non tenetur quo eum. Modi assumenda ut ut praesentium harum ducimus.
              </p>
            </div>
            <button />
          </div>
        </div>
      </div>
      <div
        className="user-img-container"
      >
        user-uploaded image goes here as button
        <button>
          <img
            src="img url goes here"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
